projects/observability/src/shared/components/bar-gauge/bar-gauge.component.ts

```diff
@@ -162,6 +165,7 @@ export interface Segment {
   label: string;
   value: number;
   color?: string;
+  tooltip?: string | TemplateRef<Segment> | TemplateRef<{ $implicit: Segment }>;
```

Comment:
@adisreyaj should I keep TemplateRef<Segment> or remove it?

Reply:
```suggestion
  tooltip?: string | TemplateRef<{ $implicit: BarSegment }>;
```

Reply:
Disagree because there are cases where we use just the segment and not the barSegment

Reply:
We are looping the `barSegments` in the template:
```html
<div *ngFor="let segment of this.barSegments"></div>
```

So this means the `segment` here is of type `BarSegment` and so the context will also be `{ $implicit: BarSegment }`.